src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupLoggingHelper.cs

```diff
@@ -81,6 +87,12 @@ internal static string GetParameterText(string prefix, string parameterName, ILi
                     {
                         sb.Append("\n");
                     }
+
+                    if (!Traits.Instance.EscapeHatches.LogTaskInputsVerbose && sb.Length >= parameterTextLimit)
```

Comment:
Do we also want to limit the number of strings outputted?

Reply:
Based on the original issue it looked like we would rather limit the raw data it uses as opposed to the number of items.

Reply:
I'd limit both. Just as you don't care about the 100,000th character, you probably don't care about the 100,000th string. You _might_ care about the last string, so ideally, you might capture the first 9k and the last 1k or something like that.

Reply:
I'll spitball 200 here, I saw one edge case(?) that was about 600 items. When building msbuild the parameters seemed to hover around 6-15.

Reply:
Consider extracting `Traits.Instance.EscapeHatches.LogTaskInputsVerbose` into a local variable outside the loop.

Reply:
First 200 items seems about right to me.

Reply:
We've seen 10k items in C++ builds.

I don't object to an item-count limit, but it makes things more complex, because a single item can go over the desired character-count limit. So you still need that, too.

Reply:
A combination of both seems like the way to go here. 200 items or 10k characters, whichever happens first will result in the rest of the results being truncated.